soccer/gameplay/skills/angle_receive.py

```diff
@@ -0,0 +1,257 @@
+            self._kick_line = robocup.Line(self.receive_point, self.get_target_point())
+
+        self._angle_facing =  target_angle_rad
+        angle_rad = self.robot.angle
+        self._angle_error = target_angle_rad - angle_rad
+
+        if self.ball_kicked:
+            receive_before_adjust = self._pass_line.nearest_point(self.robot.pos)
+        else:
+            receive_before_adjust = self.receive_point
+
+        # Make the receive point be the mouth, rather than the center of the robot.
+        # Assumes mouth of robot is at the edge.
+        self._target_pos = receive_before_adjust - robocup.Point(
+                constants.Robot.Radius * math.cos(self.robot.angle),
+                constants.Robot.Radius * math.sin(self.robot.angle))
+
+        # Code to provide slipback when receiving the ball
+        # pass_line_dir = (self._pass_line.get_pt(1) - self._pass_line.get_pt(0)).normalized()
+        # self._target_pos = actual_receive_point + pass_line_dir * constants.Robot.Radius
+
+
+        # vector pointing down the pass line toward the kicker
+        pass_dir = (self._pass_line.get_pt(0) - self._pass_line.get_pt(1)).normalized()
+
+        pos_error = self._target_pos - self.robot.pos
+        self._x_error = pos_error.dot(pass_dir.perp_ccw())
+        self._y_error = pos_error.dot(pass_dir)
+
+
+    def get_target_point(self):
+        # Gets the point the robot will be facing when receiving the ball
+        return constants.Field.TheirGoalSegment.center()
+
+    def on_exit_start(self):
+        # reset
+        self.ball_kicked = False
+
+    def execute_running(self):
+        # make sure teammates don't bump into us
+        self.robot.shield_from_teammates(constants.Robot.Radius * 2.0)
+
+        self.recalculate()
+        self.robot.face(self.robot.pos
+                + robocup.Point(math.cos(self._angle_facing), math.sin(self._angle_facing)))
+
+        if self._pass_line != None:
+            main.system_state().draw_line(self._pass_line, constants.Colors.Blue, "Pass")
+            main.system_state().draw_circle(self._target_pos, 0.03, constants.Colors.Blue, "Pass")
+        if self._kick_line != None:
+            main.system_state().draw_line(self._kick_line, constants.Colors.Red, "Shot")
+
+
+
+    def execute_aligning(self):
+        if self._target_pos != None:
+            self.robot.move_to(self._target_pos)
+
+    def on_enter_receiving(self):
+        self._shot_time = self.robot.lastKickTime()
+
+    def execute_receiving(self):
+        # Kick the ball!
+        self.robot.kick(self.kick_power)
+
+        # don't use the move_to() command here, we need more precision, less obstacle avoidance
+        pos_error = self._target_pos - self.robot.pos
+        vel = pos_error * 3.5
+        self.robot.set_world_vel(vel)
+
+        # If the shot took place, end the behavior!
+        if self._shot_time != self.robot.lastKickTime():
```

Comment:
Yup, this works for me! I didn't know that existed, i'll go ahead and replace this logic.